nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -480,6 +522,11 @@ in {
               'trusted_proxies' => ${writePhpArrary (c.trustedProxies)},
               ${optionalString (c.defaultPhoneRegion != null) "'default_phone_region' => '${c.defaultPhoneRegion}',"}
             ];
+
+            $EXTRACONFIG = json_decode(file_get_contents("${jsonFormat.generate "nextcloud-extraOptions.json" cfg.extraOptions}"), true);
```

Comment:
This creates a store-path for no reason. Any reason why we don't use `builtins.toJSON` here? Also, `json_decode` will silently fail if something goes wrong with the decode. You may want to check with `json_last_error() !== 0` and throw an error with `json_last_error_msg()` in the message.

Reply:
I recall the reason for this was that I couldn't figure out how to escape the json string produced by `builtins.toJSON` in a way that wouldn't crash php, since the only way to write a string in php appears to be quoting it with " which json also uses. Presumably the string could be escaped, but since I'm unfamiliar with the subtleties of php strings it seemed safer to do it this way.

(As I think I mentioned previously, I don't actually know much about php; the original motivation for this PR is that I needed a secretFile option and hacked one in – so while this should work, the php code may not necessarily be very elegant or idiomatic)

Reply:
Ahh you're right! While I'm not happy with this, this would escalate way to fast (I checked a few ideas, but I couldn't find a solution that's worth the effort).
However, it'd still be nice to have a safety guard here regarding how to behave if the JSON cannot be decoded by PHP. This could by implemented like this (right after the declaration of `$EXTRACONFIG`):

``` php
if (json_last_error() !== JSON_ERROR_NONE) {
  throw new \RuntimeException(sprintf("Cannot decode %s, because: %s", $filename, json_last_error_msg()));
}
```

Reply:
done (though in the nix_decode_json_file function)